molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/entity/impl/GoNLAnnotator.java

```diff
@@ -128,15 +132,21 @@ else if (inputEntity.getString(VcfRepository.REF)
 						resourceEntity.set(VcfRepository.ALT, newAltString);
 						refMatches.add(resourceEntity);
 					}
+					//situation example: input T/G, GoNL TCT/GCT
+					//we then match on T (leaving CT), and shorten the GoNL ref to T (-CT), and alt to G (-CT)
+					//now it has a match to T/G (as it should, but was not obvious due to notation)
 					else if (resourceEntity.getString(VcfRepository.REF)
 							.indexOf(inputEntity.getString(VcfRepository.REF)) == 0)
 					{
 						int postFixInputLength = resourceEntity.getString(VcfRepository.REF)
 								.substring(inputEntity.getString(VcfRepository.REF).length()).length();
+						//bugfix: matching A/G to ACT/A results in postFixInputLength=2, correctly updating ref from ACT to A,
+						//but then tries to substring the alt allele A to length -1 (1 minus 2) which is not allowed.
+						//added a check to prevent this: alt.length() > postFixInputLength ? trim the alt : change to 'n/a' because we cannot use this alt.
 						resourceEntity.set(VcfRepository.REF, resourceEntity.getString(VcfRepository.REF).substring(0,
 								(resourceEntity.getString(VcfRepository.REF).length() - postFixInputLength)));
 						String newAltString = Arrays.asList(resourceEntity.getString(ALT).split(",")).stream()
-								.map(alt -> alt.substring(0, (alt.length() - postFixInputLength)))
+								.map(alt -> alt.length() > postFixInputLength ? alt.substring(0, (alt.length() - postFixInputLength)) : "n/a")
```

Comment:
I think I aaalmost understand what's going on even!
(is N/A the standard way to write this sort of thing in vcf, not . or whatever?)